modules/install-consul/install-consul

```diff
@@ -164,14 +168,26 @@ function create_consul_install_paths {
 
 function fetch_binary {
   local -r version="$1"
-  local download_url="$2"
+  shift
+  local download_url="$1"
+  shift
+  local addtl_headers=("$@")
 
   if [[ -z "$download_url" && -n "$version" ]];  then
     download_url="https://releases.hashicorp.com/consul/${version}/consul_${version}_linux_amd64.zip"
   fi
 
+  if [[ ! -z "$addtl_headers" ]]; then
+    local headers
+
+    for value in "${addtl_headers[@]}"
+    do
+      headers+="-H \"$value\" "
```

Comment:
Slightly cleaner to make `headers` an array (declare it up above `local -a headers=()`), appending to it here via `headers+=("-H \"$value\"")`, and expanding it in the `curl` command below with `${headers[@]}`.